src/reanimated2/animation/spring.ts

```diff
@@ -137,20 +102,71 @@ export function withSpring(
       return false;
     }
 
+    function isTriggeredTwice(
+      previousAnimation: SpringAnimation | undefined,
+      animation: SpringAnimation
+    ) {
+      return (
+        previousAnimation?.type === 'spring' &&
+        previousAnimation?.toValue === animation.toValue &&
+        previousAnimation?.duration === animation.duration
+      );
+    }
+
     function onStart(
       animation: SpringAnimation,
       value: number,
       now: Timestamp,
-      previousAnimation: SpringAnimation
+      previousAnimation: SpringAnimation | undefined
     ): void {
       animation.current = value;
-      if (previousAnimation) {
-        animation.velocity =
-          previousAnimation.velocity || animation.velocity || 0;
-        animation.lastTimestamp = previousAnimation.lastTimestamp || now;
-      } else {
-        animation.lastTimestamp = now;
+      animation.startValue = value;
+
+      let mass = config.mass;
+      const triggeredTwice = isTriggeredTwice(previousAnimation, animation);
+
```

Comment:
```ts
if (triggeredTwice) {
  animation.mass = previousAnimation.mass
  ....
  return;
}
```

Reply:
The only difference is that instead of mass I set zeta, omega1 and omega0